### ISSUE TITLE
RangeMap vs NavigableMap methods (lowerEntry, floorEntry, etc)

### ISSUE THREAD

**Issue by martinm1000** [google/guava#3096]
Hi,

Is there any reason why we don't have access to some of the methods that TreeRangeMap.entriesByLowerBound offers ?

I would like an efficient method to get the lower Range key/entry from a specific existing key.
I assume I can iterate over the entry/key set in order until I get my specific key but it seems wrong  ;-)

**Comment by cpovirk** [comment]
You might be able to pull this off with `subRangeMap(...).asDescendingMapOfRanges()`, but certainly that's not as straightforward as methods like `lowerEntry`.

I'm sure we wanted to keep the API small, but I wonder if we ever considered a `RangeMaps` utility class with APIs like this. Maybe @lowasser has thoughts?

**Comment by martinm1000** [comment]
Yes, something like that will work. Just for context, I have a JTable that display the ranges and the value, and I sometimes need to read the value of the key just above the table selection, so having access to some of the navigablemap methods would be usefull.

**Comment by martinm1000** [comment]
Were still waiting for the input of @lowasser on this. If more people need something like this, the better!
you might also have other needs or comments on this too ?

**Comment by lowasser** [comment]
The intention of the API design was, in fact, to do what @cpovirk suggested.  I'm inclined to think the API is complicated enough without further bells and whistles: the intention was that you could build more complicated operations out of the narrower set we provide.
